astropy/cosmology/parameter/_core.py

```diff
@@ -7,14 +7,26 @@
 import copy
 from dataclasses import KW_ONLY, dataclass, field, fields, is_dataclass, replace
 from enum import Enum, auto
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Generic, TypeVar, overload
 
 import astropy.units as u
 
-from ._converter import _REGISTRY_FVALIDATORS, FValidateCallable, _register_validator
+from ._converter import _REGISTRY_FVALIDATORS, _register_validator
 
 if TYPE_CHECKING:
-    from collections.abc import Sequence
+    from collections.abc import Callable, Sequence
+    from typing import TypeAlias
+
+    from typing_extensions import Self
+
+    from astropy.cosmology import Cosmology
+
+    _FValidateCallable: TypeAlias = Callable[[Cosmology, Parameter, Any], _VT]  # noqa: F821
+
+
+_VT = TypeVar(
+    "_VT"
+)  # the type of the VParameter value. Required at runtime for defining `Parameter`.
```

Comment:
An attempt to keep the comment on the same line ended up splitting this over three lines.

Reply:
Refactored.